### PR TITLE
PRE application contract

### PR DESCRIPTION
**Pull request by vzotova**
Copy of nucypher/nucypher#2834 with some minor changes: deploy with proxy for tests.

### CODE REVIEW

**Review by KPrasch — COMMENTED** [review]
Do you think it's beneficial to leave `SimplePREApplication.sol` in the codebase for legacy support?

**Comment by vzotova** [comment]
> Do you think it's beneficial to leave `SimplePREApplication.sol` in the codebase for legacy support?

After we will have full application we would not maintain anything related to simple version, as far as I can see. So I'd prefer to remove it

**Review by KPrasch — APPROVED** [review]

**Review by theref — APPROVED** [review]

File: scripts/deploy_pre_application.py [hunk 1]
```diff
@@ -14,15 +14,24 @@ def main(account_id=None):
...
+        deployments_config.get("pre_hash_algorithm"),
+        deployments_config.get("pre_base_penalty"),
+        deployments_config.get("pre_penalty_history_coefficient"),
+        deployments_config.get("pre_percentage_penalty_coefficient"),
         deployments_config.get("pre_min_authorization"),
         deployments_config.get("pre_min_operator_seconds"),
+        deployments_config.get("pre_reward_duration"),
+        deployments_config.get("pre_deauthorization_duration"),
         sender=deployer,
         publish_source=deployments_config.get("verify"),
```
[the remark this answers is not in the document]
Comment (vzotova) [reply scripts/deploy_pre_application.py]: 👍 rebased over #55

[the remark this answers is not in the document]
Comment (vzotova) [reply scripts/deploy_pre_application.py]: 👍 rebased over #55

**Review by KPrasch — APPROVED** [review]
Alright - Let's merge this into `development` as discussed in Discord.

**Pull request merged**